test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,318 @@
+namespace EdgeHubRestartTester
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+    using Message = Microsoft.Azure.Devices.Client.Message;
+
+    class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("EdgeHubRestartTester");
+
+        static uint restartCount = 0;
+        static long messageCount = 0;
+        static long directMethodCount = 0;
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Guid batchId = Guid.NewGuid();
+            Logger.LogInformation($"Starting Edge Hub Restart Tester ({batchId}) with the following settings:\r\n{Settings.Current}");
+
+            Logger.LogInformation($"Load gen delay start for {Settings.Current.TestStartDelay}.");
+            await Task.Delay(Settings.Current.TestStartDelay);
+
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
+
+            ServiceClient iotHubServiceClient = null;
+            ModuleClient msgModuleClient = null;
+            ModuleClient dmModuleClient = null;
+            try
+            {
+                iotHubServiceClient = ServiceClient.CreateFromConnectionString(Settings.Current.ServiceClientConnectionString);
+
+                if (Settings.Current.MessageEnable)
+                {
+                    msgModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.MessageTransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+                    msgModuleClient.OperationTimeoutInMilliseconds = Settings.Current.SdkRetryTimeout;
+                }
+
+                if (Settings.Current.DirectMethodEnable)
+                {
+                    dmModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.MessageTransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+                }
+
+                DateTime testStart = DateTime.UtcNow;
+                DateTime testExpirationTime = testStart + Settings.Current.TestDuration;
+                Dictionary<string, Task<Tuple<DateTime, HttpStatusCode>>> taskList = new Dictionary<string, Task<Tuple<DateTime, HttpStatusCode>>>();
```

Comment:
I can't find the usage, please remove it.

Reply:
Good catch, that's an artifact from my first implementation.